app/models/preference.rb

```diff
@@ -5,6 +5,9 @@
 class Preference < ApplicationRecord
   belongs_to :applicant
   belongs_to :position
+
+  validates_presence_of :priority, :applicant, :position
+  validates :priority, numericality: { only_integer: true, less_than_or_equal_to: 3 }
```

Comment:
Perhaps we should make this an enum too? something like `%i[low medium high]` or at least ensure that it can't be negative?